### ISSUE TITLE
prefer-stateless-function does not detect when using sub-render() functions.

### ISSUE THREAD

**Issue by jseminck** [yannickcr/eslint-plugin-react#1235]
Not the most clear title, let me explain with an example. In our code, we often split up the `render()` function into smaller functions when it starts getting too big. For example:

```js
export default class MyComponent extends React.Component {
    static propTypes = {
        someArray: React.PropTypes.array.isRequired,
    };

    render() {
        return (
            <div>
                <div>
                   ... some static jsx goes here ...
                </div>
                {this.renderArray()}
            </ul>
        );
    }

    renderArray() {
        return this.props.myArray.map((item, index) => {
            return <Item key={index} {...item} />;
        });
    }
}
```

The `prefer-stateless-function` rule doesn't seem to detect this to be converted into a stateless function component, even though in theory it could totally be a stateless function component.

I did not look at the code but I assume it is because it detects another function that is different from `render()` ?

I understand this style isn't very popular in the React community (at least, I have not seen it in any other open-source projects). So it seems to be something that's specific to my team. Therefore I am not sure if the plugin should detect this at all? And even if it should, I am not sure how the `--fix` option would work for it...

**Comment by jseminck** [comment]
Additionally, it also doesn't detect this. This seems already like a more common use case to me?

```js
export default class MyComponent extends React.Component {
    static propTypes = {
        myAction: PropTypes.func.isRequired,
    }

    onClick() {
        this.props.myAction();
    }

    render() {
        return (
            <button onClick={() => {}}>Click me!</button>
        );
    }
}
```

**Comment by ljharb** [comment]
I think we'd need that behavior under a new option - but I like it, because `renderFoo` methods should always be SFCs.

Your latter example shouldn't be an SFC, it should be a  `class` with a constructor-bound instance method. Using `::` inline there is incorrect.

**Comment by jseminck** [comment]
I'll look into this: reporting methods that return JSX other than the `render()` method, should be reported to be a stateless functional component.
